app/src/components/market/sections/market_buy/market_buy.tsx

```diff
@@ -152,7 +154,7 @@ const MarketBuyWrapper: React.FC<Props> = (props: Props) => {
     cost.isZero() ||
     isBuyAmountGreaterThanBalance ||
     amount.isZero() ||
-    !hasEnoughAllowance
+    hasEnoughAllowance !== Ternary.True
```

Comment:
The button will be disabled if the user doesn't have enough allowance, or if we don't know how much they have.